tutorials/01_introductory/lifecycle.py

```diff
@@ -0,0 +1,288 @@
+to highlight some neat features and best-practices using Matplotlib.
+
+.. currentmodule:: matplotlib
+
+.. note::
+
+    This tutorial is based off of
+    `this excellent blog post <http://pbpython.com/effective-matplotlib.html>`_
+    by Chris Moffitt. It was transformed into this tutorial by Chris Holdgraf.
+
+A note on Object-Oriented vs Pyplot
+===================================
+
+Matplotlib has two interfaces. The first is based on MATLAB and uses
+a state-based interface. This is encapsulated in the :mod:`pyplot`
+module. See the :ref:`pyplot tutorials <sphx_glr_tutorials_01_introductory_lifecycle.py>`
+for a more in-depth look at the pyplot interface.
+
+The second interface is an an object-oriented (OO) interface.
+In this case, we utilize an instance of :class:`axes.Axes` in order to
+render visualizations on an instance of :class:`figure.Figure`.
+
+Most of the terms are straightforward but the main thing to remember
+is that:
+
+* the Figure is the final image that may contain 1 or more axes.
+* The Axes represent an individual plot.
+
+We call methods that do the plotting directly from the Axes, which gives
+us much more flexibility and power in customizing our plot. See the
+:ref:`object-oriented examples <api_examples>` for many examples of how
+this approach is used.
+
+.. note::
+
+   In general, try to use the object-oriented interface over the pyplot
+   interface.
+
+Our data
+========
+
+We'll use the data from the post from which this tutorial was derived.
+It contains sales information for a number of companies.
+"""
+
+# sphinx_gallery_thumbnail_number = 10
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.ticker import FuncFormatter
+from string import ascii_lowercase
+
+data = {'Barton LLC': 109438.5,
+        'Frami, Hills and Schmidt': 103569.59,
+        'Fritsch, Russel and Anderson': 112214.71000000001,
+        'Jerde-Hilpert': 112591.42999999999,
+        'Keeling LLC': 100934.29999999999,
+        'Koepp Ltd': 103660.54000000001,
+        'Kulas Inc': 137351.95999999996,
+        'Trantow-Barrows': 123381.38000000003,
+        'White-Trantow': 135841.99000000002,
+        'Will LLC': 104437.60000000002}
+group_data = list(data.values())
+group_names = list(data.keys())
+group_mean = np.mean(group_data)
+
+###############################################################################
+# Getting started
+# ===============
+#
+# This data is naturally visualized as a barplot, with one bar per
+# group. To do this with the object-oriented approach, we'll first generate
+# an instance of :class:`figure.Figure` and
+# :class:`axes.Axes`. The Figure is like a canvas, and the Axes
+# is a part of that canvas on which we will make a particular visualization.
+#
+# .. note::
+#
+#    Figures can have multiple axes on them. For information on how to do this,
+#    see the :ref:`Tight Layout tutorial <sphx_glr_tutorials_02_intermediate_tight_layout_guide.py>`.
```

Comment:
That doesn't seem to work on my end. I tried:

```
:ref:`my text
<my_label>`
```
and
```
:ref:`my text\
<my_label>`
```
but neither seemed to work...